astro/src/content/docs/get-started/run-in-the-cloud/github-actions.mdx

```diff
@@ -0,0 +1,456 @@
+
+To see what is happening visually, run the code below.
+
+```bash
+npx playwright test --project=chromium --ui
+```
+
+You can see that Playwright runs a browser, clicks buttons, and fills in forms.
+
+<img src="/img/docs/get-started/run-in-the-cloud/github_actions/playwright.png" alt="Playwright test UI." role="bottom-cropped" />
+
+## Write A Test For Playwright
+
+The code for the test Playwright runs is in the file `test.spec.js`.
+
+<RemoteCode url={frontmatter.codeRoot + "/app/tests/test.spec.js"}
+  lang="js" />
+
+The code should mostly be self-explanatory, but there are a few things to note:
+- `await` is used frequently as most operations on the page are asynchronous.
+- `waitForLoadState('networkidle')` is used to wait for pages to load. When this article was written, `.waitForNavigation()` and `.waitForURL()` failed.
+- Tests start with `expect()`, such as `expect(emailText).toBe('richard@example.com')`. Without this function, your tests won't test anything.
+- To add a new test of your own, write a new function like `test('Test login', async ({ page }) => {`.
+
+To add Playwright to your own project, follow [their guide](https://playwright.dev/docs/intro). Running `npm init playwright` will create a configuration file and sample test you can use as a starting point.
+
+In the configuration file `playwright.config.js`, you can set which browsers to use (we used `--project=chromium` earlier), whether to start your app before running the tests, and which directory contains test files.
+
+## Test Your App In A GitHub Action
+
+Now that you know how to test your app after a change to it or FusionAuth, let's automate the process in GitHub. Your goal is to have a test run whenever committing to your main branch. GitHub will email you if the test fails.
+
+Open the file `.github/workflows/test.yaml`. Line 5 currently reads as follows.
+
+```yaml
+      - main_RENAME_THIS_TO_ENABLE_TEST
+```
+
+Change the line to the following.
+
+```yaml
+      - main
+```
+
+You've renamed the trigger that causes this action to execute, from a nonsense name to the main branch of your application.
+
+By default, GitHub prevents action workflows from running on forked repositories. To grant permission for the workflow to execute, navigate to the Actions tab and click on the <InlineUIElement>I understand my workflows, go ahead and enable them</InlineUIElement> button on the repository. Save, commit in Git, and push to GitHub, and the test will execute.
+
+You can browse to the Actions tab in your GitHub repository to see the test run.
+
+<img src="/img/docs/get-started/run-in-the-cloud/github_actions/passedtest.png" alt="Passed GitHub action test." role="bottom-cropped" />
+
+To test that the test works, change the email in `test.spec.js` to `wrong@example.com` and push to GitHub again. You can now verify that the test does detect a broken login.
```

Comment:
should be more clear:

change:
expect(emailText).toBe('richard@example.com');

to:
expect(emailText).toBe('wrong@example.com');